### PR TITLE
Fix quoting when using inheritance as well

### PR DESCRIPTION
**Pull request by lcobucci**
Related to #6402 and #6404

### CODE REVIEW

File: lib/Doctrine/ORM/Persisters/Entity/AbstractEntityInheritancePersister.php [hunk 1]
```diff
@@ -82,16 +82,18 @@ protected function getSelectColumnSQL($field, ClassMetadata $class, $alias = 'r'
...
      * @param string $tableAlias
      * @param string $joinColumnName
+     * @param string $quotedColumnName
+     *
      * @param string $type
      *
      * @return string
      */
-    protected function getSelectJoinColumnSQL($tableAlias, $joinColumnName, $type)
+    protected function getSelectJoinColumnSQL($tableAlias, $joinColumnName, $quotedColumnName, $type)
```
Comment (lcobucci) [inline lib/Doctrine/ORM/Persisters/Entity/AbstractEntityInheritancePersister.php]: Theoretically this is as minor BC break, should we add to `UPGRADING.md`?

Comment (guilhermeblanco) [inline lib/Doctrine/ORM/Persisters/Entity/AbstractEntityInheritancePersister.php]: Not a valid BC break since we do not allow custom persisters.

**Comment by guilhermeblanco** [comment]
`develop` is way ahead of the game and honestly any change around quoting is ok since it'll all changed already.

**Comment by Ocramius** [comment]
@guilhermeblanco just a note: rebasing on this will likely be annoying. Still, the fix is valid IMO, and it shall be merged in `master`

File: lib/Doctrine/ORM/Persisters/Entity/AbstractEntityInheritancePersister.php (hunk 1 above)
Comment (Ocramius) [inline lib/Doctrine/ORM/Persisters/Entity/AbstractEntityInheritancePersister.php]: Indeed, change is fine here

**Review by Ocramius — APPROVED** [review]
LGTM :ship:

**Pull request merged**